services/admin/backend/api/middleware/permissions.go

```diff
@@ -0,0 +1,172 @@
+package middleware
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/golang-jwt/jwt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+type AccessDetails struct {
+	UserId string
+	Permissions []string
+}
+
+// Requesting Party Token
+type RPT struct {
+	// Scopes Client authorization scopes (essentially the permission strings)
+	Scopes []string `json:"scopes"`
+
+	// ResourceId Client authorization resource id
+	ResourceId string `json:"rsid"`
+
+	// ResourceName Client authorization resource name
+	ResourceName string `json:"rsname"`
+}
```

Comment:
Should we keep it the permissions array as `scopes` or maybe this can be changed to `permissions`?
Can we have more informative camelCased names for the others keys?

Reply:
I don't know honestly. I mean, technically they are scopes in regards to Keycloak. I think it's fine to leave this as `scopes` because this struct is really confined to this and shouldn't be used elsewhere, maybe I can make it private or something. I also don't want there to be confusion between `RPT.Permissions` and `AccessDetails.Permissions`. The whole point of this permissions middleware is really to return an `AccessDetails` object with the `userId` and list of `permissions`. I adjusted the json attribute name in 45509a7 to be more descriptive.

Reply:
That's makes sense to leave it as `scopes`.